### PR TITLE
Update list object options, add slab encryption key and make bucket optional

### PR DESCRIPTION
**Pull request by peterjan**
This PR updates `ListObjectOptions`, making `bucket` optional and adding `slabEncryptionKey`. It removes an endpoint and allows fetching objects cross buckets, which we use for migration failure alerts.

### CODE REVIEW

File: bus/routes.go [hunk 1]
```diff
@@ -1352,10 +1351,11 @@ func (b *Bus) objectsListHandlerPOST(jc jape.Context) {
 	if jc.Decode(&req) != nil {
 		return
 	}
-	if req.Bucket == "" {
-		req.Bucket = api.DefaultBucketName
+	limit := -1
```
Comment (peterjan) [inline bus/routes.go:1354]: Defaulting `offset` and `limit` in a F/U. Looks like this went under the radar for some time.

File: stores/sql/main.go [hunk 2]
```diff
@@ -1522,6 +1522,7 @@ func ObjectMetadata(ctx context.Context, tx Tx, bucket, key string) (api.Object,
 	om, err := tx.ScanObjectMetadata(tx.QueryRow(ctx, fmt.Sprintf(`
 		SELECT %s
 		FROM objects o
+		INNER JOIN buckets b ON b.id = o.db_bucket_id
```
[the remark this answers is not in the document]
Comment (peterjan) [reply stores/sql/main.go:1525]: We want to add `Bucket` to the metadata so I had to change `SelectObjectMetadataExpr` to include `b.name`. That is why in one or two occasions I have to `INNER JOIN` on buckets to ensure the bucket alias is defined. I originally did not want to update the object expression but it's much cleaner to do so (, at the cost of having to join with buckets).

File: stores/metadata_test.go [hunk 3]
```diff
@@ -3255,67 +3269,68 @@ func TestContractSizes(t *testing.T) {
...
-		if objs[i].Size != int64(i)+1 {
-			t.Fatal("unexpected object size", objs[i].Size, i+1)
-		}
-		if objs[i].Health != 1.0 {
-			t.Fatal("unexpected object health", objs[i].Health)
-		}
-	}
-}
+//  : use this as a list test
+// func TestObjectsBySlabKey(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (peterjan) [reply stores/metadata_test.go:3273]: oversight, thanks

File: stores/metadata_test.go
[the remark this answers is not in the document]
Comment (peterjan) [reply stores/metadata_test.go:1]: It's here https://github.com/SiaFoundation/renterd/pull/1517/files#diff-ed88e5314575f8e61b39911dbb329eb8fe821f5ed2d9b1e1f953376e5b02e768R3457 . I understand that's quite minimal but in this particular case I felt we don't need the table test.

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**